### DISCUSSION TITLE
keg-only dependency installations are slightly confusing...

### DISCUSSION THREAD

**Discussion by rdp** [Homebrew/discussions#1415] [Everyday usage]
It's a bit...confusing to see the following:

```
$ brew install postgresql
...
==> Installing dependencies for postgresql: icu4c
==> Installing postgresql dependency: icu4c
==> Pouring icu4c--69.1.big_sur.bottle.tar.gz
==> Caveats
icu4c is keg-only, which means it was not symlinked into /usr/local,
because macOS provides libicucore.dylib (but nothing else).

If you need to have icu4c first in your PATH, run:
  echo 'export PATH="/usr/local/opt/icu4c/bin:$PATH"' >> ~/.zshrc
  echo 'export PATH="/usr/local/opt/icu4c/sbin:$PATH"' >> ~/.zshrc

For compilers to find icu4c you may need to set:
  export LDFLAGS="-L/usr/local/opt/icu4c/lib"
  export CPPFLAGS="-I/usr/local/opt/icu4c/include"

For pkg-config to find icu4c you may need to set:
  export PKG_CONFIG_PATH="/usr/local/opt/icu4c/lib/pkgconfig"

==> Summary
🍺  /usr/local/Cellar/icu4c/69.1: 259 files, 72.8MB
==> Installing postgresql
...
```

It makes me wonder if I need to do those things in order to have postgres "work at all".  Wonder if there's a way to clarify it more?

**Comment by carlocab** [comment]
If a keg-only formula was installed as a dependency, it should just work. If you find need to tweak things like environment variables to get a keg-only dependency to work with a formula, please file an issue in Homebrew/core. That's a bug. (Or, better, open a PR.)

If you're using a keg-only formula directly (and not as a dependency of another formula), then you may or may not want to set those things (depending on what works for you).

We're happy to review pull requests that improve the messaging here. The relevant piece of code can be found in [this method](https://github.com/Homebrew/brew/blob/master/Library/Homebrew/caveats.rb#L50). Perhaps a link to the [FAQ](https://docs.brew.sh/FAQ#what-does-keg-only-mean) might help, but I suspect the FAQ doesn't actually answer your questions here. You're also welcome to open a PR to improve that FAQ; the place you'll want to edit it is [here](https://github.com/Homebrew/brew/blob/master/docs/FAQ.md#what-does-keg-only-mean).

**Comment by rdp** [comment]
It has been fixed since to not show dependency caveats at the end or something like that, thanks for your help!
